job-server/test/spark.jobserver/TestJarFinder.scala

```diff
@@ -1,11 +1,24 @@
 package spark.jobserver
 
+import java.nio.file.Paths
+
 trait TestJarFinder {
   val versionRegex = """(\d\.\d+).*""".r
   val version = scala.util.Properties.versionNumberString match { case versionRegex(d) => d }
-  val testJarDir = "../job-server-tests/target/scala-" + version + "/"
-  val testJar = {
-    val allJars = new java.io.File(testJarDir).listFiles.toSeq.filter { file =>
+
+  def testJarDir: String = "job-server-tests/target/scala-" + version + "/"
```

Comment:
I have a fear that someone might forget to add the version in a future impl of TesJarFinder, ExtrasTestJarFinder does the same thing.